static/js/libs/arrays.js

```diff
@@ -0,0 +1,23 @@
+function arraysEqual(oldArray, newArray) {
+  if (oldArray === newArray) {
+    return true;
+  }
+  if (
+    oldArray === null || newArray === null ||
+    (oldArray.length !== newArray.length)) {
+    return false;
+  }
+
+  oldArray.sort();
```

Comment:
Yeah so before the sorting the function checks "are they the same array?" if not is either of them null, or are the lengths different? Then is sorts both of them and checks each element in the array, the side effects or sorting shouldn't matter as after the sort it's checking on an item by item basis.

Reply:
By side effects I mean you are chaning the arrays you are 'checking'.

```js
a = [3, 2, 1];
b = [2, 1, 3];
arraysEqual(a, b); // true
a; // [1, 2, 3]
b; // [1, 2, 3]
```

Reply:
Unless this is expected and you want to make the arrays equal?

Reply:
Oh I see! I'll add a param to the function to indicate whether you want sorting or not

Reply:
@Lukewh if `arraysEquals` is meant only to check if 2 arrays contain same elements (and from its usage it seems it is) it should never change the arrays given as params. If sorting is required to allow checking for items in any order it should be done on clones of original arrays, so that objects given to the function would not be affected by it.

Reply:
of course, sorry baby brain.